neonix/io/nixio.py

```diff
@@ -0,0 +1,270 @@
+# Copyright (c) 2014, German Neuroinformatics Node (G-Node)
+#                     Achilleas Koutsou <achilleas.k@gmail.com>
+#
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted under the terms of the BSD License. See
+# LICENSE file in the root of the Project.
+
+from __future__ import absolute_import
+
+import numpy as np
+import quantities as pq
+
+from neo.io.baseio import BaseIO
+from neo.core import Block, Segment, RecordingChannelGroup
+
+try:
+    import nix
+except ImportError:
+    raise ImportError("Failed to import NIX (NIXPY not found). "
+                      "The NixIO requires the Python bindings for NIX.")
+
+
+attribute_mappings = {"name": "name",
+                      "description": "definition"}
+container_mappings = {"segments": "groups",
+                      "recordingchannelgroups": "sources"}
+
+
+class NixIO(BaseIO):
+    """
+    Class for reading and writing NIX files.
+    """
+
+    is_readable = False  # for now
+    is_writable = True
+
+    supported_objects = [Block, Segment, RecordingChannelGroup]
+    readable_objects = []
+    writeable_objects = [Block, Segment, RecordingChannelGroup]
+
+    name = "NIX"
+    extensions = ["h5"]
+    mode = "file"
+
+    def __init__(self, filename):
+        """
+        Initialise IO instance and NIX file.
+
+        :param filename: full path to the file
+        """
+        BaseIO.__init__(self, filename=None)
+        self.filename = filename
+        if self.filename:
+            self.nix_file = nix.File.open(self.filename, nix.FileMode.Overwrite)
+
+    def __del__(self):
+        self.nix_file.close()
+
+    def write_block(self, neo_block, cascade=True):
+        """
+        Convert ``neo_block`` to the NIX equivalent and write it to file.
+        If ``cascade`` is True write all the block's child objects as well.
+
+        :param neo_block: Neo block to be written
+        :param cascade: save all child objects (default: True)
+        """
+        nix_name = neo_block.name
+        nix_type = "neo.block"
+        nix_definition = neo_block.description
+        nix_block = self.nix_file.create_block(nix_name, nix_type)
+        nix_block.definition = nix_definition
+        if neo_block.rec_datetime:
+            # Truncating timestamp to seconds
+            nix_block.force_created_at(int(neo_block.rec_datetime.timestamp()))
```

Comment:
You're right. It looks like the simplest way to get the equivalent timestamp in Python2 is to subtract from epoch directly.